src/main/org/deidentifier/arx/io/ImportAdapterJDBC.java

```diff
@@ -1,309 +1,419 @@
+package org.deidentifier.arx.io;
+
+import org.deidentifier.arx.DataType;
+
+import java.io.IOException;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Import adapter for JDBC
+ * <p>
+ * This adapter can import data from JDBC sources. The source itself is
+ * described by an appropriate {@link ImportConfigurationJDBC} object.
+ *
+ * @author Karol Babioch
+ * @author Fabian Prasser
+ */
+public class ImportAdapterJDBC extends ImportAdapter {
+
+    /**
+     * The configuration describing the CSV file being used.
+     */
+    private ImportConfigurationJDBC config;
+
+    /**
+     * ResultSet containing rows to return.
+     *
+     * @see {@link #next()}
+     */
+    private ResultSet resultSet;
+
+    /**
+     * JDBC statement.
+     */
+    private Statement statement;
+
+    /**
+     * Indicates whether there is another row to return.
+     */
+    private boolean hasNext;
+
+    /**
+     * Indicates whether the first row has already been returned
+     * <p>
+     * The first row contains the name of the columns. Depending upon whether
+     * the name of the column has been assigned explicitly, this is either the
+     * value of the table itself, the value defined by the user.
+     */
+    private boolean headerReturned;
+
+    /**
+     * Number of rows that need to be processed in total.
+     *
+     * @see {@link #getProgress()}
+     */
+    private int totalRows;
+
+    /**
+     * Creates a new instance of this object with given configuration.
+     *
+     * @param config {@link #config}
+     * @throws IOException In case of communication errors with JDBC
+     * @todo Fix IOException
+     */
+    protected ImportAdapterJDBC(ImportConfigurationJDBC config) throws IOException {
+
+        super(config);
+        this.config = config;
+
+        /* Preparation work */
+        indexes = getIndexesToImport();
+        dataTypes = getColumnDatatypes();
+
+        try {
+
+            /* Used to keep track of progress */
+            statement = config.getConnection().createStatement();
+
+            String tableName = config.getTable();
+            String schemaName = config.getSchemaName();
+            String dbName = config.getDbName();
+            DatabaseMetaData meta = config.getConnection().getMetaData();
+            String productName = meta.getDatabaseProductName().toLowerCase();
+
+            String afterTableName = "";
+
+            String databaseType = "";
+
+            if (productName.contains("oracle")) {
+                databaseType = "oracle";
+            } else if (productName.contains("db2")) {
+                databaseType = "db2";
+            } else if (productName.startsWith("microsoft sql server")) {
+                databaseType = "sqlserver";
+            } else if (productName.contains("mysql")) {
+                databaseType = "mysql";
+            } else {
+                databaseType = "other";
+            }
+
+            switch (databaseType) {
```

Comment:
Please also check that schemaName and dbName are not null. It would also be great to support cases in which only schemaName!=null or only dbName!=null.